redislock.go

```diff
@@ -98,8 +124,16 @@ func (c *Client) Obtain(ctx context.Context, key string, ttl time.Duration, opt
 	}
 }
 
-func (c *Client) obtain(ctx context.Context, key, value string, ttl time.Duration) (bool, error) {
-	return c.client.SetNX(ctx, key, value, ttl).Result()
+func (c *Client) obtain(ctx context.Context, keys []string, value string, ttl time.Duration) (bool, error) {
+	setArgs := []string{}
+	for _, k := range keys {
+		setArgs = append(setArgs, k, value)
+	}
+	res, err := luaMSetNXEX.Run(ctx, c.client, setArgs, ttl.Milliseconds()).Result()
+	if err != nil {
+		return false, fmt.Errorf("msetnx failed: %s", err)
+	}
+	return res == int64(len(keys)), nil
```

Comment:
As I mentioned in msetnx script above - pretty much a no-op.

The script could just return boolean on its own. No need to expose count of set-expires to the caller.

Doesn't prevent us from merging and then polishing it ourselves.

Reply:
updated error handling strategy. Please let me know what you think.